editor/connections_dialog.cpp

```diff
@@ -994,11 +998,11 @@ void ConnectionsDock::_tree_item_selected() {
 		connect_button->set_text(TTR("Connect..."));
 		connect_button->set_icon(get_theme_icon(SNAME("Instance"), SNAME("EditorIcons")));
 		connect_button->set_disabled(true);
-	} else if (_is_item_signal(*item)) {
+	} else if (_get_item_type(*item) == TREE_ITEM_TYPE_SIGNAL) {
 		connect_button->set_text(TTR("Connect..."));
 		connect_button->set_icon(get_theme_icon(SNAME("Instance"), SNAME("EditorIcons")));
 		connect_button->set_disabled(false);
-	} else {
+	} else if (_get_item_type(*item) == TREE_ITEM_TYPE_CONNECTION) {
```

Comment:
Should there still be the default branch that disables the button? In case if the class item is selected, I guess.